### PR TITLE
Add JPEG support

### LINKED ISSUE #2: JPG support

### PR DESCRIPTION
**Pull request by eventualbuddha**
Closes #2

### CODE REVIEW

**Comment by kAworu** [comment]
Hello @eventualbuddha,

Thanks a lot for this PR. I'm taking a look at it and overall it look good to me.

As a library, I think we should avoid printing on stderr (on jpeg error). Once that is gone I'll be happy to merge and make a new release!

**Comment by eventualbuddha** [comment]
@kAworu I removed the stderr print. I had pretty much just copied all the code from quirc so I didn't really think too much about the printing, but you're right. I also updated the README to note that JPEG is now supported.

File: src/node_quirc_decode.c [hunk 1]
```diff
@@ -311,3 +316,83 @@ nq_load_png(struct quirc *q, const uint8_t *img, size_t img_len)
...
+	jmp_buf env;
+};
+
+static void nq_output_message(struct jpeg_common_struct *com)
+{
+	struct nq_jpeg_error *err = (struct nq_jpeg_error *)com->err;
+	char buf[JMSG_LENGTH_MAX];
+
+	err->base.format_message(com, buf);
+	fprintf(stderr, "JPEG error: %s\n", buf);
```
Comment (eventualbuddha) [inline src/node_quirc_decode.c]: Whoops, I missed this one.

Comment (eventualbuddha) [reply src/node_quirc_decode.c]: Okay, fixed.

**Pull request merged**